test/deprecated.jl

```diff
@@ -1,6 +1,12 @@
-using FiniteDifferences: rand_tangent
+# Test struct for `rand_tangent` and `difference`.
+struct Foo
+    a::Float64
+    b::Int
+    c::Any
+ end
+rand_tangent(args...) = @test_deprecated FiniteDifferences.rand_tangent(args...)
```

Comment:
that's pretty cool - you are defining a new function here, which calls the original one?

Reply:
Yeah, It is pretty lazy.

```suggestion

# to avoid deprecation spam (and actually test deprecations) we will define a wrapper `rand_tangent` function for testing
rand_tangent(args...) = @test_deprecated FiniteDifferences.rand_tangent(args...)
```